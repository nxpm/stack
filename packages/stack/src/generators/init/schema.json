{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "id": "Init",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    },
    "apiName": {
      "type": "string",
      "description": "Name of the API project",
      "default": "api"
    },
    "mobileName": {
      "type": "string",
      "description": "Name of the Mobile project",
      "default": "mobile"
    },
    "webName": {
      "type": "string",
      "description": "Name of the Web project",
      "default": "web"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "alias": "d"
    }
  },
  "required": ["name"]
}
