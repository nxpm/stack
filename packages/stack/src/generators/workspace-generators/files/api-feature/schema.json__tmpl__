{
  "cli": "nx",
  "id": "api-feature",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "Library name",
      "$default": {
        "$source": "argv",
        "index": 0
      }
    },
    "appName": {
      "type": "string",
      "description": "Application name",
      "default": "api"
    }
  },
  "required": ["name", "appName"]
}
