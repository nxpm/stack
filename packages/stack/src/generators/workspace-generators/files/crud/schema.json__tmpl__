{
  "cli": "nx",
  "id": "crud",
  "type": "object",
  "properties": {
    "model": {
      "type": "string",
      "description": "Name of the model (in kebab-case), for example: company-address",
      "$default": {
        "$source": "argv",
        "index": 0
      }
    },
    "plural": {
      "type": "string",
      "description": "Name of the model (in kebab-case), for example: company-addresses",
      "$default": {
        "$source": "argv",
        "index": 1
      }
    }
  },
  "required": ["model", "plural"]
}
