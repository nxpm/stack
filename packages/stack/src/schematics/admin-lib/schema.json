{
  "$schema": "http://json-schema.org/draft-07/schema",
  "id": "AdminLib",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the library?",
      "pattern": "^[a-zA-Z]{1}.*$"
    },
    "type": {
      "description": "The type of library",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 1
      },
      "x-prompt": {
        "message": "Which type of library would you like to create?",
        "type": "list",
        "items": [
          {
            "value": "assets",
            "label": "Assets library"
          },
          {
            "value": "data-access",
            "label": "Data access library"
          },
          {
            "value": "feature",
            "label": "Feature library"
          },
          {
            "value": "style",
            "label": "Style Library"
          },
          {
            "value": "ui",
            "label": "UI Library"
          },
          {
            "value": "util",
            "label": "Utility library"
          }
        ]
      }
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "alias": "d"
    },
    "publishable": {
      "type": "boolean",
      "default": false,
      "description": "Generate a publishable library."
    },
    "buildable": {
      "type": "boolean",
      "default": false,
      "description": "Generate a buildable library."
    },
    "prefix": {
      "type": "string",
      "format": "html-selector",
      "description": "The prefix to apply to generated selectors.",
      "alias": "p"
    }
  },
  "required": ["name"]
}
