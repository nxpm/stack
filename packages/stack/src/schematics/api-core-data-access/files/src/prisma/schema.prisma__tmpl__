datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native"]
}

model User {
  id                   String           @id @default(cuid())
  createdAt            DateTime         @default(now())
  updatedAt            DateTime         @default(now()) @updatedAt
  role                 Role
  developer            Boolean          @default(false)
  email                String           @unique
  username             String           @unique
  password             String?
  firstName            String?
  lastName             String?
  avatarUrl            String?
  location             String?
  phone                String?
  bio                  String?
}

enum Role {
  Admin
  User
}

