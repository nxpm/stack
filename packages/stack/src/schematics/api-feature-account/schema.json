{
  "$schema": "http://json-schema.org/draft-07/schema",
  "id": "ApiFeatureUser",
  "title": "",
  "type": "object",
  "properties": {
    "appName": {
      "type": "string",
      "description": "Name of the app to connect to",
      "alias": "a"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "alias": "d"
    }
  },
  "required": []
}
