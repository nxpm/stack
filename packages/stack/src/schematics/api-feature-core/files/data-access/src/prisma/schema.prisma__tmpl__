datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native"]
}

model User {
  id        String   @id @default(cuid())
  createdAt DateTime @default(now())
  updatedAt DateTime @default(now()) @updatedAt
  role      Role
  developer Boolean  @default(false)
  username  String   @unique
  password  String?
  firstName String?
  lastName  String?
  avatarUrl String?
  location  String?
  phone     String?
  bio       String?
  emails    Email[]  @relation("Email_belongsTo_Owner")
}

enum Role {
  Admin
  User
}

model Email {
  id            String    @id @default(cuid())
  createdAt     DateTime  @default(now())
  updatedAt     DateTime  @default(now()) @updatedAt
  email         String    @unique
  public        Boolean   @default(false)
  primary       Boolean   @default(false)
  verified      Boolean   @default(false)
  verifyToken   String?
  verifyExpires DateTime?
  owner         User      @relation(name: "Email_belongsTo_Owner", fields: [ownerId], references: [id])
  ownerId       String
  // Enable when this is resolved https://github.com/prisma/prisma/issues/3387
  // @@unique([ownerId, primary])
}
