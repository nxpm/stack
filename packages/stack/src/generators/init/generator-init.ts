import { formatFiles, Tree } from '@nrwl/devkit'
import { generatorApi } from '@nxpm/api'
import { normalizeOptions, workspaceConfig } from '@nxpm/common'
import { join } from 'path'

import { InitGeneratorSchema } from './schema'

export async function generatorInit(host: Tree, options: InitGeneratorSchema) {
  // Prepare options
  const normalizedOptions = normalizeOptions<InitGeneratorSchema>(host, options)

  if (!normalizedOptions.skipApi) {
    // Run API
    await generatorApi(host, {
      ...normalizedOptions,
      name: normalizedOptions.appNameApi,
      type: normalizedOptions.appTypeApi,
    })
  }

  if (!normalizedOptions.skipMobile) {
    // Run Mobile
    // await generatorMobile(host, { name: options.mobileName, type: options.mobileType })
  }

  if (!normalizedOptions.skipWeb) {
    // Run Web
    // await generatorWeb(host, { name: options.webName, type: options.webType })
  }

  workspaceConfig(host, normalizedOptions, join(__dirname, 'files'))
  await formatFiles(host)
}
