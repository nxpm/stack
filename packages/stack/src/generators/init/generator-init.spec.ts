import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing'
import { Tree } from '@nrwl/devkit'

import { generatorInit } from './generator-init'

import { InitGeneratorSchema } from './schema'

describe('init generator', () => {
  let appTree: Tree
  const options: InitGeneratorSchema = { name: 'test' }

  beforeEach(() => {
    appTree = createTreeWithEmptyWorkspace()
  })

  it('should run successfully', async () => {
    await generatorInit(appTree, options)
    // const config = readProjectConfiguration(appTree, 'test')
    const files = appTree.children('.')
    console.log(files)
    expect(appTree.children('.')).toBeDefined()
  })
})

describe('init generator: custom options', () => {
  let appTree: Tree
  const options: InitGeneratorSchema = {
    name: 'test',
    apiName: 'server',
    mobileName: 'native',
    webName: 'frontend',
  }

  beforeEach(() => {
    appTree = createTreeWithEmptyWorkspace()
  })

  it('should run successfully', async () => {
    await generatorInit(appTree, options)
    // const config = readProjectConfiguration(appTree, 'test')
    const files = appTree.children('.')
    console.log(files)
    expect(appTree.children('.')).toBeDefined()
  })
})
