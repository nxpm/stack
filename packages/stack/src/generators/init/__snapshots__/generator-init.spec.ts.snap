// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init generator should run successfully with default options 1`] = `
Array [
  "src",
  "tsconfig.app.json",
  "tsconfig.json",
  ".eslintrc.json",
  "jest.config.js",
  "tsconfig.spec.json",
]
`;

exports[`init generator should run successfully with default options 2`] = `
Array [
  "app",
  "assets",
  "environments",
  "main.ts",
]
`;

exports[`init generator should run successfully with default options 3`] = `
Array [
  "app.module.ts",
]
`;

exports[`init generator should run successfully with default options 4`] = `
Object {
  "projects": Object {
    "api": Object {
      "name": "test-api",
      "type": "nest",
    },
    "mobile": Object {
      "name": "mobile",
      "type": "ionic-angular",
    },
    "web": Object {
      "name": "web",
      "type": "angular",
    },
  },
}
`;

exports[`init generator should run successfully with default options 5`] = `
Array [
  "workspace.json",
  ".prettierrc",
  "package.json",
  "nx.json",
  "tsconfig.base.json",
  "jest.config.js",
  "jest.preset.js",
  "apps",
  ".eslintrc.json",
  "libs",
  ".env",
  ".env.example",
  ".gitignore",
  "Dockerfile",
  ".dockerignore",
  "docker-compose.yml",
  ".prettierignore",
  "nxpm.json",
  "README.md",
]
`;

exports[`init generator should run successfully with default options 6`] = `
"{
  \\"name\\": \\"test-name\\",
  \\"dependencies\\": {
    \\"@kikstart-playground/graphql-intercom\\": \\"1.4.1\\",
    \\"@nestjs/common\\": \\"^7.0.0\\",
    \\"@nestjs/config\\": \\"^0.5.0\\",
    \\"@nestjs/core\\": \\"^7.0.0\\",
    \\"@nestjs/graphql\\": \\"7.11.0\\",
    \\"@nestjs/jwt\\": \\"^7.2.0\\",
    \\"@nestjs/passport\\": \\"^7.1.5\\",
    \\"@nestjs/platform-express\\": \\"^7.0.0\\",
    \\"@nestjs/serve-static\\": \\"2.2.2\\",
    \\"@prisma/client\\": \\"2.27.0\\",
    \\"apollo-server-express\\": \\"^2.18.0\\",
    \\"bcryptjs\\": \\"^2.4.3\\",
    \\"class-transformer\\": \\"^0.3.1\\",
    \\"class-validator\\": \\"^0.12.2\\",
    \\"cookie-parser\\": \\"1.4.5\\",
    \\"fs-extra\\": \\"10.0.0\\",
    \\"graphql\\": \\"^15.3.0\\",
    \\"graphql-subscriptions\\": \\"1.2.1\\",
    \\"graphql-tools\\": \\"^6.2.3\\",
    \\"graphql-type-json\\": \\"^0.3.2\\",
    \\"joi\\": \\"^17.2.1\\",
    \\"passport\\": \\"^0.4.1\\",
    \\"passport-jwt\\": \\"^4.0.0\\",
    \\"reflect-metadata\\": \\"^0.1.13\\",
    \\"rxjs\\": \\"~6.6.3\\",
    \\"tslib\\": \\"^2.0.0\\"
  },
  \\"devDependencies\\": {
    \\"@nestjs/schematics\\": \\"^7.0.0\\",
    \\"@nestjs/testing\\": \\"^7.0.0\\",
    \\"@nrwl/eslint-plugin-nx\\": \\"12.5.8\\",
    \\"@nrwl/jest\\": \\"12.5.8\\",
    \\"@nrwl/linter\\": \\"12.5.8\\",
    \\"@nrwl/nest\\": \\"12.5.8\\",
    \\"@nrwl/node\\": \\"12.5.8\\",
    \\"@types/fs-extra\\": \\"9.0.12\\",
    \\"@types/jest\\": \\"26.0.8\\",
    \\"@typescript-eslint/eslint-plugin\\": \\"4.19.0\\",
    \\"@typescript-eslint/parser\\": \\"4.19.0\\",
    \\"eslint\\": \\"7.22.0\\",
    \\"eslint-config-prettier\\": \\"8.1.0\\",
    \\"jest\\": \\"27.0.3\\",
    \\"prisma\\": \\"2.27.0\\",
    \\"supertest\\": \\"6.1.3\\",
    \\"ts-jest\\": \\"27.0.3\\"
  },
  \\"prisma\\": {
    \\"schema\\": \\"libs/test-api/core/data-access/src/prisma/schema.prisma\\"
  },
  \\"scripts\\": {
    \\"build\\": \\"yarn build:web && yarn prisma:generate && yarn build:test-api\\",
    \\"build:test-api\\": \\"nx build test-api --prod\\",
    \\"dev:services\\": \\"docker-compose up\\",
    \\"dev:test-api\\": \\"nx serve test-api\\",
    \\"docker:build\\": \\"docker build . -t proj/test-api\\",
    \\"docker:push\\": \\"docker push proj/test-api\\",
    \\"docker:run\\": \\"docker run -it -p 8000:3000 proj/test-api\\",
    \\"prisma:apply\\": \\"yarn prisma:format && yarn prisma:db-push\\",
    \\"prisma:db-push\\": \\"yarn prisma db push\\",
    \\"prisma:format\\": \\"yarn prisma format\\",
    \\"prisma:generate\\": \\"yarn prisma generate\\",
    \\"prisma:migrate\\": \\"yarn prisma migrate\\",
    \\"prisma:seed\\": \\"yarn nx workspace-generator prisma-seed\\",
    \\"prisma:studio\\": \\"yarn prisma studio\\",
    \\"setup\\": \\"yarn nx workspace-generator workspace-setup\\",
    \\"start\\": \\"yarn prisma:db-push && node dist/apps/api/main.js\\",
    \\"test:ci\\": \\"yarn prisma:apply && yarn e2e api-e2e\\"
  }
}"
`;

exports[`init generator: custom options should run successfully with custom options 1`] = `
Array [
  "src",
  "tsconfig.app.json",
  "tsconfig.json",
  ".eslintrc.json",
  "jest.config.js",
  "tsconfig.spec.json",
]
`;

exports[`init generator: custom options should run successfully with custom options 2`] = `
Array [
  "app",
  "assets",
  "environments",
  "main.ts",
]
`;

exports[`init generator: custom options should run successfully with custom options 3`] = `
Array [
  "app.module.ts",
]
`;

exports[`init generator: custom options should run successfully with custom options 4`] = `
Object {
  "projects": Object {
    "api": Object {
      "name": "server",
      "type": "nest",
    },
    "mobile": Object {
      "name": "native",
      "type": "ionic-angular",
    },
    "web": Object {
      "name": "frontend",
      "type": "angular",
    },
  },
}
`;
